src/main/java/io/nats/client/impl/NatsConsumerContext.java

```diff
@@ -56,6 +60,13 @@ public class NatsConsumerContext implements ConsumerContext, SimplifiedSubscript
         defaultDispatcher = new AtomicReference<>();
         lastConsumer = new AtomicReference<>();
         if (unorderedConsumerInfo != null) {
+        	if (unorderedConsumerInfo != null) {
+                //Fail fast on push consumers as all operation on ConsumerContext expect a pull consumer.
+                //If we don't fail fast, calls to consume() or iterat() will hang with the server returning: code=409, message='Consumer is push based'
+                ConsumerConfiguration consumerConfig = unorderedConsumerInfo.getConsumerConfiguration();
+                if ( consumerConfig.getDeliverSubject() != null )
+                    throw new JetStreamApiException(io.nats.client.api.Error.instance(409, "Consumer is push based. ConsumerContext only supports pull consumers."));
+        	}
```

Comment:
1. I'm not 100% no but... I'm pretty sure it was discussed in depth during simplification design to not fast fail. Maybe it should change. Also maybe it should be done in the caller. But for now just leave it.
2. JetStreamApiException are reserved for JS errors that come from the server. For errors like this that come from the client, make a NatsJetStreamClientError

Reply:
OK, moved to another branch. You can in that branch.